test/server/listener_manager_impl_test.cc

```diff
@@ -1512,6 +1523,31 @@ TEST_F(ListenerManagerImplWithRealFiltersTest, FreebindListenerEnabled) {
   }
 }
 
+// Set the resolver to the default IP resolver. The address resolver logic is unit tested in
+// resolver_impl_test.cc.
+TEST_F(ListenerManagerImplWithRealFiltersTest, AddressResolver) {
+  const std::string yaml = TestEnvironment::substitute(R"EOF(
+    name: AddressResolverdListener
+    address:
+      socket_address: { address: 127.0.0.1, port_value: 1111, resolver_name: envoy.mock.resolver }
+    filter_chains:
+    - filters:
+  )EOF",
+                                                       Network::Address::IpVersion::v4);
+
+  MockAddressResolver mock_resolver;
+  EXPECT_CALL(mock_resolver, name()).WillRepeatedly(Return("envoy.mock.resolver"));
```

Comment:
I would probably just setup a default ON_CALL in the mock for this one. (EXPECT_CALL/WillRepeatedly is a bit of an anti-pattern)